### ISSUE TITLE
ArrayBuffer -> Blob?

### ISSUE THREAD

**Issue by alch-emi** [purescript-web/purescript-web-file#22]
Hello!

I was wondering if there's any way to turn an `ArrayBuffer` into a `Blob`.  If I understand correctly, the JavaScript implementation of `new Blob(...)` allows for passing an array of blob parts, which can include `ArrayBuffer` (js) and `ArrayBufferView` (js).  I don't see anything similar in the [`Web.File.Blob`][1] docs though, so I was wondering if there's a recommended way to do this.

My use case, in case of XY problems:  I'm downloading an encrypted file using [`purescript-fetch`][2], then decrypting it using [`purescript-subtlecrypto`][3], which yields an `ArrayBuffer` (which I have a known `image/*` MIME for) that I want to display to the user.  I know I could use data urls, but I'd like to avoid the memory/cpu overhead if possible.

Also, I'm a bit of a purescript novice, so apologies in advance if I've missed something that's obvious to more experienced users.

Thanks for your help!

[1]: https://pursuit.purescript.org/packages/purescript-web-file/4.0.0/docs/Web.File.Blob
[2]: https://pursuit.purescript.org/packages/purescript-fetch/1.1.2
[3]: https://pursuit.purescript.org/packages/purescript-subtlecrypto/0.0.1

**Comment by garyb** [comment]
No, I don't think you've missed anything - or if you have, I have too :smile: - I think we are indeed lacking a constructor for that case.

There's probably a couple of variations that are missing - if we wanted to support creating a `Blob` from multiple parts we'd have to introduce a `BlobPart` type and some coercion functions. Would also need to change the FFI implementation a bit if we wanted to support the `endings` option.
